pkg/cli/completion/completion.go

```diff
@@ -0,0 +1,103 @@
+package completion
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/k3s-io/k3s/pkg/version"
+
+	"github.com/urfave/cli"
+)
+
+func Run(ctx *cli.Context) error {
+	if ctx.NArg() < 1 {
+		return fmt.Errorf("must provide a valid SHELL argument")
+	}
+	shell := ctx.Args()[0]
+	completetionScript, err := genCompletionScript(shell)
+	if err != nil {
+		return err
+	}
+	autoFile := "/etc/rancher/" + version.Program + "/autocomplete"
```

Comment:
Why are we writing this out to a file in `/etc`? This seems problematic for users who may install K3s using sudo but primarily interact with it as a non-root user. There's no reason to write out a file containing the completions ;the convention for other apps in the ecosystem is to just have the completion command write the script out to stdout; adding the commands to the shell is accomplished by something like `source <(kubectl completion bash)` rather than sourcing a prewritten file in /etc.

See: https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-bash-linux/

Reply:
The problem is when first creating, we always output some info to stderr 
```
derek@degion:~/rancher/k3s$ ./dist/artifacts/k3s completion bash
INFO[0000] Acquiring lock file /home/derek/.rancher/k3s/data/.lock 
INFO[0000] Preparing data dir /home/derek/.rancher/k3s/data/d0a880c3a35a101faf3d6bc9cc1f16e712964f8894ea60292de2a7c1f2234f8f 
```
I could blackhole any stderr into `/dev/null` but then if the command is wrong, it won't let the user know.

Reply:
Okay I have converted those 2 lines to `debug` level, so they don't appear when running `k3s completion` I have also changed it so that by default `k3s completion bash` just writes to stdout, no `/etc/rancher/k3s/`. Additionally, I added a `-i` flag that simply adds the source line to a bashrc/zshrc.

Reply:
@briandowns I think this is why he converted them to debug, but I think perhaps a better approach would be to set the log level to ERROR for the completion command to prevent them from being output, if they're creating problems. Honestly they go to STDERR not STDOUT, and should only pop up once when it's initially run, so it should be fine?

You might also consider having the self-extracting wrapper include the completion functionality itself, to avoid having to extract out the actual binary just to provide the completion scripts.

Reply:
If logs are going to stdout, switching them to stderr could also be an easy fix, in addition to or instead of changing the log level.

Reply:
OKay another commit, I reverted the levels back to Info, and now I am just making sure that for k3s-completion in the multicall, we set the default log level to Error.